scripts/globals/utils.lua

```diff
@@ -112,6 +112,25 @@ function utils.bind(func, ...)
     end
 end
 
+-- Merges two array-style tables into a new one
+-- Creates and returns a new iterable table containing all elements from both input arrays `a` and `b`,
+---@nodiscard
+---@param a table
+---@param b table
+---@return table
+function utils.mergeArrays(a, b)
```

Comment:
To me, `merge` implies that they're getting mixed as `abababab` - this is more: `appendArray(a, b)` -> a|b